lib/assert.js

```diff
@@ -106,13 +107,23 @@ function fail(actual, expected, message, operator, stackStartFn) {
       operator = '!=';
   }
 
-  innerFail({
+  if (message instanceof Error) throw message;
+
```

Comment:
☝️ Should use the util for `isError` (internal reference) for this since it avoids the error of errors from other realms.

Reply:
Only changing this here is likely not the best way. It would have to be changed throughout assert in total or not at all. It is also sad that the util function is not a safe check as the `Symbol.toStringTag` could be set and `isError` would happily return true in some weird cases.

Reply:
>Only changing this here is likely not the best way.

Ah, didn't know this was consistent with the rest of `assert`. A follow up would be best.

> It is also sad that the util function is not a safe check as the Symbol.toStringTag could be set and isError would happily return true in some weird cases.

Oh I thought the new `isNativeError` was more robust.

Reply:
Oh, I missed the `isNativeError` somehow. We have multiple places in core where we should likely switch to this check.